mathics/doc/common_doc.py

```diff
@@ -952,24 +988,327 @@ class Documentation:
     """
 
     def __init__(self):
-        self.chapter_class = DocChapter
-        self.chapters: List[DocChapter] = []
-        self.doc_class = DocumentationEntry
-        self.doc_dir = settings.DOC_DIR
-        self.guide_section_class = DocGuideSection
-        self.guide_sections: List[DocGuideSection] = []
-        self.part_class = DocPart
-        self.parts: list[DocPart] = []
+        # This is a way to load the default classes
+        # without defining these attributes as class
+        # attributes.
+        self._set_classes()
```

Comment:
Using this mechanism seems better that passing the classes as initialization parameters. Then, each class can have its own strategy to set the attributes.